DDD.Functions/SubmitSessionFeedback.cs

```diff
@@ -0,0 +1,91 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using System.Net;
+using System.Threading.Tasks;
+using System.Net.Http;
+using System;
+using System.Linq;
+using DDD.Functions.Extensions;
+using DDD.Core.AzureStorage;
+
+namespace DDD.Functions
+{
+    public static class SubmitSessionFeedback
+    {
+        [FunctionName("SubmitSessionFeedback")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "post", Route = null)]
+            HttpRequestMessage req,
+            ILogger log,
+            [BindConferenceConfig]
+            ConferenceConfig conference,
+            [BindKeyDatesConfig]
+            KeyDatesConfig keyDates,
+            [BindFeedbackConfig]
+            FeedbackConfig feedbackConfig,
+            [BindSessionsConfig]
+            SessionsConfig sessionsConfig
+            )
+        {
+            var feedback = await req.Content.ReadAsAsync<SessionFeedbackRequest>();
+
+            // Within feedback window
+            if (keyDates.Before(x => x.FeedbackAvailableFromDate) || keyDates.After(x => x.FeedbackAvailableToDate))
+            {
+                log.LogWarning("Attempt to access SubmitSessionFeedback endpoint outside of allowed window of {start} -> {end}.", keyDates.FeedbackAvailableFromDate, keyDates.FeedbackAvailableToDate);
+                return new StatusCodeResult((int)HttpStatusCode.NotFound);
+            }
+
+            // Valid feedback 
+            if ((feedback.Rating < 1 || feedback.Rating > 5) && string.IsNullOrWhiteSpace(feedback.Likes) && string.IsNullOrWhiteSpace(feedback.ImprovementIdeas))
```

Comment:
are we doing 1-5? I personally thing 1-5 is effectively meaningless, had good experience with -1,0,1 (so negative, neutral, positive), but happy to go with the consensus.

1-5 has issues with range compression/j curves (what really is the difference between a 5 star talk and a 4 star talk - ?) 1-5 also has issues with people needing to do more thinking, which leads to more dropoffs. 


Link if you're super bored :-)  https://pdfs.semanticscholar.org/9430/f3a76ce04cb9de4b74fd231e954c7386c96c.pdf?_ga=2.96423196.1646251308.1563496672-279858019.1563496672

Reply:
Removed the rating check so it can be any number :)